### DISCUSSION TITLE
Cannot combine Nextjs and Slider of Material UI

### DISCUSSION THREAD

**Discussion by teppei-kogure** [react-hook-form/react-hook-form#2087] [Q&A] [answered]
I wrote the following script to combine Next.JS and rhf, but the following error occurs. Can this be avoided?

 `import React, { useState } from "react";
import { useForm, Controller } from "react-hook-form";
import Header from "./Header";
import {
  TextField,
  RadioGroup,
  FormControlLabel,
  ThemeProvider,
  Radio,
  createMuiTheme,
  Slider
} from "@material-ui/core";
function App() {
  const { handleSubmit, reset, control, errors, getValues } = useForm({ defaultValues });
  const [data, setData] = useState(null);
  const handleChange = (event, newData) => {
    setData(newData)
  };
  const marks = []
  for (let i = 0; i <= 10; i++) {
    marks.push({
        value: i,
        label: String(i)
    })
  }
return (
    <ThemeProvider theme={theme}>
      <form onSubmit={handleSubmit(data => setData(data))}className="form">
        <Header  />
        <div>
                <Controller
                     name="manner_points"
                     control={control}
                     onChange={handleChange}
                     as={<Slider valueLabelDisplay="auto" marks={marks} max={10} step={1} />}
                />
        </div>

        <ButtonsResult {...{ data, reset, defaultValues }} />
      </form>
      {/* <style jsx>{`
      `}</style> */}
    </ThemeProvider>
  );
}
export default App;` 

But, When operating the Slider, I get the following error
`Material-UI: A component is changing the controlled value state of Slider to be uncontrolled.
Elements should not switch from uncontrolled to controlled (or vice versa).
Decide between using a controlled or uncontrolled Slider element for the lifetime of the component.
The nature of the state is determined during the first render, it's considered controlled if the value is not `undefined`.
`
I think it's a state-related problem, but I don't know how to fix it. Please tell me if there is any means

**Comment by bluebill1049** [comment, marked as answer]
sounds like you are missing `defaultVlaue`

**Reply by teppei-kogure** [reply]
Which side of the defult value is wrong? Or how should I fix it?

**Reply by bluebill1049** [reply]
at the `Controller`

**Reply by teppei-kogure** [reply]
I checked folling this url
https://codesandbox.io/s/react-hook-form-v6-controller-24gcl?file=/src/index.jsx:3473-3927

 and I missed state .

**Reply by bluebill1049** [reply]
it's using `defaultValues` at useForm
```
useForm({
  defaultValues
});
```
so you have must miss at the `defaultValues`

**Reply by teppei-kogure** [reply]
Thank you very much. That helps a lot
